metaflow/plugins/pypi/conda_decorator.py

```diff
@@ -353,6 +353,13 @@ def is_attribute_user_defined(self, name):
     def flow_init(
         self, flow, graph, environment, flow_datastore, metadata, logger, echo, options
     ):
+        # NOTE: This makes sure that using conda_base marks all steps as executing with conda,
+        # without relying on --environment conda setting decospecs.
+        # Important for extensions implementing custom virtual environments
+        from metaflow import decorators
+
+        decorators._attach_decorators(flow, ["conda"])
```

Comment:
do we need a symmetric change for `pypi_base` too? if not, can you add the reasoning as a code comment?

Reply:
pypi_base already does this for itself so no change needed on that side. I'll refine the above comment a bit to clarify.